### PR TITLE
Badge: refactor using our own Icon component

### PR DESCRIPTION
**Pull request by driesd**
### Description

This PR refactors our `Badge` component to use our `Icon` component instead of a custom styled span.

### Breaking changes

None.

### CODE REVIEW

File: src/components/badge/Badge.js [hunk 1]
```diff
@@ -25,6 +26,20 @@ class Badge extends PureComponent {
...
   };
 
+  renderIcon = () => {
+    const { color, icon, inverse } = this.props;
+
+    return (
+      <Icon
+        className={theme['icon']}
+        color={color === 'neutral' ? 'teal' : color}
+        tint={inverse ? 'lightest' : 'darkest'}
```
Comment (kevinwaelkens) [inline src/components/badge/Badge.js]: Q: Don't we want to be able to also pass down a tint? Or will this *never* be the case?

Comment (driesd) [reply src/components/badge/Badge.js]: According to the design specs it should be neutral darkest :)

Comment (kevinwaelkens) [reply src/components/badge/Badge.js]: But now it's going to be `teal` `darkest` when it's `neutral` right :P?

Comment (driesd) [reply src/components/badge/Badge.js]: That is correct and is expected behavior. We did this for contrast reasons. You can check the design specs on Zeplin if you want: https://app.zeplin.io/project/599af68db2f4978a56dec839/screen/5a537ce8a7baf1d15409c86e

**Pull request merged**